src/mkdocs_bibtex/utils.py

```diff
@@ -309,3 +312,55 @@ def tempfile_from_url(name, url, suffix):
     raise RuntimeError(
         f"Couldn't successfully download the url: {url}"
     )  # pragma: no cover
+
+def tempfile_from_zotero(name: str, url: str, suffix: str) -> str:
+    """Download bibfile from the Zotero API."""
+    log.debug(f"Downloading {name} from Zotero at {url}")
+    bib_contents = ""
+    url = sanitize_zotero_query(url)
+    for page_num in range(999):
+        for _ in range(3):
+            try:
+                response = requests.get(url)
+                if response.status_code != 200:
+                    msg = f"Couldn't download the url: {url}.\nStatus Code: {response.status_code}"
+                    raise RuntimeError(msg)
+                break
+            except requests.exceptions.RequestException:  # pragma: no cover
+                pass
+
+        bib_contents += response.text
+        try:
+            url = response.links["next"]["url"]
+        except KeyError:
+            log.debug(f"Downloaded {page_num}(s) from {url}")
+            break
+    else:
+        log.debug(f"Exceeded the maximum number of pages. Found: {page_num} pages")
+    with tempfile.NamedTemporaryFile(mode="wt", encoding="utf-8", suffix=suffix, delete=False) as file:
+        file.write(bib_contents)
+        log.info(f"{name} downloaded from URL {url} to temporary file ({file})")
```

Comment:
The log probably doesn't need to be within the context. Probably better to let the context clean up before logging here in case there's a problem cleaning up.

```suggestion
    log.info(f"{name} downloaded from URL {url} to temporary file ({file})")
```